ambry-tools/src/main/java/com.github.ambry/tools/admin/ConcurrencyTestTool.java

```diff
@@ -519,7 +519,7 @@ public void onCompletion(Void result, Exception exception) {
 
     MetricsCollector(long measurementIntervalInSecs, String operation) {
       this.measurementIntervalInSecs = measurementIntervalInSecs;
-      requestLatencies = new ArrayList<Long>();
+      requestLatencies = new CopyOnWriteArrayList<>();
```

Comment:
So, it seems like `CopyOnWriteArrayList.sort()` still takes a lock for the duration of the sort (that prevents any concurrent `add()`s), so it might be better to just use an external lock for synchronization, so that we don't incur the copy-on-write cost.